Makefile

```diff
@@ -361,82 +279,133 @@ generate-go-conversions-kcp: $(CONVERSION_GEN)
+	GOLANGCI_LINT_EXTRA_ARGS=--fix $(MAKE) lint
+
+apidiff: $(GO_APIDIFF) ## Check for API differences
+	$(GO_APIDIFF) $(shell git rev-parse origin/main) --print-compatible
+
+.PHONY: format-tiltfile
+format-tiltfile: ## Format the Tiltfile
+	./hack/verify-starlark.sh fix
+
+.PHONY: verify
+verify: ## Run all verify-* targets
+	./hack/verify-doctoc.sh
+	$(MAKE) verify-boilerplate
+	$(MAKE) verify-shellcheck
+	$(MAKE) verify-tiltfile
+	$(MAKE) verify-modules
+	$(MAKE) verify-gen
+	$(MAKE) verify-conversions
+	$(MAKE) verify-docker-provider
+
+.PHONY: verify-modules
+verify-modules: modules  ## Verify go modules are up to date
+	@if !(git diff --quiet HEAD -- go.sum go.mod $(TOOLS_DIR)/go.mod $(TOOLS_DIR)/go.sum $(TEST_DIR)/go.mod $(TEST_DIR)/go.sum); then \
+		git diff; \
+		echo "go module files are out of date"; exit 1; \
+	fi
+	@if (find . -name 'go.mod' | xargs -n1 grep -q -i 'k8s.io/client-go.*+incompatible'); then \
+		find . -name "go.mod" -exec grep -i 'k8s.io/client-go.*+incompatible' {} \; -print; \
+		echo "go module contains an incompatible client-go version"; exit 1; \
+	fi
+
+.PHONY: verify-gen
+verify-gen: generate  ## Verfiy go generated files are up to date
+	@if !(git diff --quiet HEAD); then \
+		git diff; \
+		echo "generated files are out of date, run make generate"; exit 1; \
+	fi
+
+.PHONY: verify-conversions
+verify-conversions: $(CONVERSION_VERIFIER)  ## Verifies API expected conversion are in place
+	$(CONVERSION_VERIFIER)
+
+.PHONY: verify-boilerplate
+verify-boilerplate: ## Verify boilerplate text exists in each file
+	./hack/verify-boilerplate.sh
+
+.PHONY: verify-shellcheck
+verify-shellcheck: ## Verify shell files
+	./hack/verify-shellcheck.sh
+
+.PHONY: verify-tiltfile
+verify-tiltfile: ## Verify Tiltfile format
+	./hack/verify-starlark.sh
 
-## --------------------------------------
-## Modules
-## --------------------------------------
+.PHONY: verify-docker-provider
+verify-docker-provider:
+	@echo "Verifying CAPD"
+	cd $(CAPD_DIR); $(MAKE) verify
 
-.PHONY: modules
-modules: ## Runs go mod to ensure modules are up to date.
-	go mod tidy
-	cd $(TOOLS_DIR); go mod tidy
-	cd $(TEST_DIR); go mod tidy
+.PHONY: verify-book-links
+verify-book-links:
+	$(MAKE) -C docs/book verify
 
 ## --------------------------------------
-## Docker
+## Binaries
 ## --------------------------------------
 
-.PHONY: docker-pull-prerequisites
-docker-pull-prerequisites:
-	docker pull docker.io/docker/dockerfile:1.1-experimental
-	docker pull $(GO_CONTAINER_IMAGE)
-	docker pull gcr.io/distroless/static:latest
+##@ build:
+
+.PHONY: clusterctl
+clusterctl: ## Build the clusterctl binary
+	go build -trimpath -ldflags "$(LDFLAGS)" -o $(BIN_DIR)/clusterctl sigs.k8s.io/cluster-api/cmd/clusterctl
+
+.PHONY: managers
+managers: ## Run all manager-* targets
+	$(MAKE) manager-core
+	$(MAKE) manager-kubeadm-bootstrap
+	$(MAKE) manager-kubeadm-control-plane
```

Comment:
What about adding `ALL_MANAGERS` variable and use `$(addprefix manager-,$(ALL_MANAGERS))`

Reply:
+1 Added.